config/config.go

```diff
@@ -1,25 +1,36 @@
 package config
 
 import (
+	"fmt"
 	"os"
 	"strconv"
+	"strings"
+
+	"github.com/joho/godotenv"
 )
 
 // Config holds the application configuration
 type Config struct {
-	DiceAddr      string
-	ServerPort    string
-	RequestLimit  int // Field for the request limit
-	RequestWindow int // Field for the time window in seconds
+	DiceAddr       string
+	ServerPort     string
+	RequestLimit   int64    // Field for the request limit
+	RequestWindow  float64  // Field for the time window in float64
+	AllowedOrigins []string // Field for the allowed origins
 }
 
 // LoadConfig loads the application configuration from environment variables or defaults
 func LoadConfig() *Config {
+	err := godotenv.Load()
+	if err != nil {
+		fmt.Println("Warning: .env file not found, falling back to system environment variables.")
+	}
+
 	return &Config{
-		DiceAddr:      getEnv("DICE_ADDR", "localhost:7379"), // Default Dice address
-		ServerPort:    getEnv("SERVER_PORT", ":8080"),        // Default server port
-		RequestLimit:  getEnvInt("REQUEST_LIMIT", 1000),      // Default request limit
-		RequestWindow: getEnvInt("REQUEST_WINDOW", 60),       // Default request window in seconds
+		DiceAddr:       getEnv("DICE_ADDR", "localhost:7379"),                                  // Default Dice address
+		ServerPort:     getEnv("SERVER_PORT", ":8080"),                                         // Default server port
+		RequestLimit:   getEnvInt("REQUEST_LIMIT", 1000),                                       // Default request limit
+		RequestWindow:  getEnvFloat64("REQUEST_WINDOW", 60),                                    // Default request window in float64
+		AllowedOrigins: getEnvArray("ALLOWED_ORIGINS", []string{"*", "http://localhost:8080"}), // Default allowed origins
```

Comment:
We'll need to later update it, so that only traffic from 1 origin is allowed.

Reply:
yes, i think we can remove only the default params.